web_widget_char_switchcase/static/src/js/web_widget_char_switchcase.js

```diff
@@ -17,19 +20,11 @@ odoo.define('web_widget_char_switchcase', function (require) {
             if (!_.contains(this.transformations, this.current_transformation))
                 console.error(this.current_transformation + ' case unknown');
         },
-        initialize_content: function() {
-            var res = this._super();
-            var self = this;
-            if(this.$input) {
-                this.$input.keyup(function(){
-                    var old_val = self.$input.val();
-                    if (!old_val)
-                        return;
-                    var new_val = self.current_transformation_handler(old_val);
-                    self.$input.val(new_val);
-                });
-            }
-            return res;
+        key_up_handler: function ($event) {
+            var $input = $($event.target);
```

Comment:
don't you have `this.$input.val()`?
And you should do the check here on `if this.$input.val().trim()`.
also, is not `$event` but `event` as is not a jq object

Reply:
Yes, I realized that I had the correct `this` after writing that.
Done in https://github.com/OCA/web/pull/1079/commits/63e2623b69d22a27b224fe80586e2fef4a9ac99e